### ISSUE TITLE
`sqitch.conf` file permission mask

### ISSUE THREAD

**Issue by fluca1978** [theory/sqitch#839]
Using `sqitch` `1.4.1` from CPAN.
I noted that `sqitch.conf` has default read permission for everyone:

```
% sqitch --version
sqitch (App::Sqitch) v1.4.1

 % ls -l sqitch.conf
-rw-rw-r-- 1 luca luca 222 nov 19 09:41 sqitch.conf
```

Apparently the application is working even if permissions are set for only the user, thus:

```
% chmod 600 sqitch.conf
```

Shouldn't this be default permission mask since `sqitch.conf` could possibly contain connection passwords?

**Comment by theory** [comment]
Never thought about it. It defaults to whatever the current directory's umask is.

**Comment by theory** [comment]
[Looks like](https://github.com/bestpractical/config-gitlike/blob/a065320af482d33142116f8ecf67606bdb9da48d/lib/Config/GitLike.pm#L1127C1-L1128C1) Config::GitLike, which creates the file, does not have any support for custom permissions:

```pl
    sysopen(my $fh, "${filename}.lock", O_CREAT|O_EXCL|O_WRONLY)
        or die "Can't open ${filename}.lock for writing: $!\n";
```

Looks like it would require another attribute on the object, similar to `encoding`, to specify permissions to pass as the fourth argument to [`sysopen`](https://perldoc.perl.org/functions/sysopen), something like:

```pl
sysopen(my $fh, "${filename}.lock", O_CREAT|O_EXCL|O_WRONLY, $self->permissions || 0666)
```

If they added such an option I'd be happy to use it in Sqitch.

**Comment by fluca1978** [comment]
Would it have `sqitch` to automatically adjust permissions at first read  a possible solution?

**Comment by fluca1978** [comment]
@theory  PTAL at <https://github.com/fluca1978/sqitch/commit/d4ab60e57d217220f10af1878d1f94b45febcbf2>. Seems very simple solution to me, but works for both adding and editing configuration.
Clearly it makes extra work, since it checks that the configuration file(s) have the right permissions every time they are loaded, but at least it adjusts the permissions as soon as possible.
Clearly, there is no solution against a user that deliberately taint his own environment changing the permissions outside of the `sqitch` application.

**Comment by fluca1978** [comment]
> Config::GitLike operates on a whole [chain of config files](https://sqitch.org/docs/manual/sqitch-config/#files), including, potentially, files not owned by the current user. I think this should be limited to those owned by the current user, no?
> 

Good point! I guess, rethinking about this, that probably it does not have any sense to force permissions, since `sqitch` could use a globally defined file that, in turn, contains a set of target credentials that are shared, hence readable by different users.
Therefore, the only thing that could make sense, according to me, is to warn the user if the configuration file has world read access, letting her to change the permissions if appropriate.


> How does Git deal with config file permissions?

I need to think about this and document, I have no idea so far.
